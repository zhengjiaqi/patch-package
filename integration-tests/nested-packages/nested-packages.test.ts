import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "nested-packages",
  shouldProduceSnapshots: true,
})
