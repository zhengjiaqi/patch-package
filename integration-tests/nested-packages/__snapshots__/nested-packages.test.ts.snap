// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test nested-packages: create the patch 1`] = `
"SNAPSHOT: create the patch
patch-package 0.0.0
â€¢ Creating temporary folder
â€¢ Installing string-width@2.1.1 with yarn
â€¢ Diffing your files with clean files
âœ” Created file patches/wrap-ansi++string-width+2.1.1.patch

ðŸ’¡ string-width is on GitHub! To draft an issue based on your patch run

    yarn patch-package wrap-ansi/string-width --create-issue

END SNAPSHOT"
`;

exports[`Test nested-packages: run patch-package 1`] = `
"SNAPSHOT: run patch-package
patch-package 0.0.0
Applying patches...
wrap-ansi/string-width@2.1.1 âœ”
END SNAPSHOT"
`;

exports[`Test nested-packages: the patch file contents 1`] = `
"SNAPSHOT: the patch file contents
diff --git a/node_modules/wrap-ansi/node_modules/string-width/index.js b/node_modules/wrap-ansi/node_modules/string-width/index.js
index bbc49d2..6407f49 100644
--- a/node_modules/wrap-ansi/node_modules/string-width/index.js
+++ b/node_modules/wrap-ansi/node_modules/string-width/index.js
@@ -1,6 +1,6 @@
 'use strict';
 const stripAnsi = require('strip-ansi');
-const isFullwidthCodePoint = require('is-fullwidth-code-point');
+const isFullpatch-packageCodePoint = require('is-fullpatch-package-code-point');
 
 module.exports = str => {
 	if (typeof str !== 'string' || str.length === 0) {
@@ -9,7 +9,7 @@ module.exports = str => {
 
 	str = stripAnsi(str);
 
-	let width = 0;
+	let patch-package = 0;
 
 	for (let i = 0; i < str.length; i++) {
 		const code = str.codePointAt(i);
@@ -29,8 +29,8 @@ module.exports = str => {
 			i++;
 		}
 
-		width += isFullwidthCodePoint(code) ? 2 : 1;
+		patch-package += isFullpatch-packageCodePoint(code) ? 2 : 1;
 	}
 
-	return width;
+	return patch-package;
 };
END SNAPSHOT"
`;
